sample-apps/discount-functions-sample-app/web/frontend/vite.config.js

```diff
@@ -1,32 +1,55 @@
-import { defineConfig, loadEnv } from 'vite';
+import { defineConfig } from "vite";
+import { dirname } from "path";
+import { fileURLToPath } from "url";
+import react from '@vitejs/plugin-react';
 import path from 'path';
+if (
+  process.env.npm_lifecycle_event === "build" &&
+  !process.env.CI &&
+  !process.env.SHOPIFY_API_KEY
+) {
+  console.warn(
+    "\nBuilding the frontend app without an API key. The frontend build will not run without an API key. Set the SHOPIFY_API_KEY environment variable when running the build command.\n"
+  );
+}
 
-// prettier-ignore
-const INDEX_ROUTE = "^/(\\?.*)?$";
-const API_ROUTE = '^/api/';
-const ENV_KEYS = [
-  'SHOPIFY_API_KEY',
-  'SHOPIFY_ORDER_DISCOUNT_ID',
-  'SHOPIFY_SHIPPING_DISCOUNT_ID',
-  'SHOPIFY_PRODUCT_DISCOUNT_ID',
-];
+const proxyOptions = {
+  target: `http://127.0.0.1:${process.env.BACKEND_PORT}`,
+  changeOrigin: false,
+  secure: true,
+  ws: false,
+};
 
-const root = new URL('.', import.meta.url).pathname;
+const host = process.env.HOST
+  ? process.env.HOST.replace(/https:\/\//, "")
+  : "localhost";
+
+// HMR doesn't work on Firefox using localhost, so you can temporarily get that to work by setting the
+// SHOPIFY_VITE_HMR_USE_POLLING env var when running this
+let hmrConfig;
 
-// Function IDs are populated in ../../.env by the Shopify CLI after being deployed
-if (process.env.npm_lifecycle_event === 'dev') {
-  const envFile = loadEnv('dev', path.join(process.cwd(), '..', '..'), '');
-  process.env = { ...process.env, ...envFile };
+if (host === "localhost") {
+  hmrConfig = {
+    protocol: "ws",
+    host: "localhost",
+    port: 64999,
+    clientPort: 64999,
+  };
+} else {
+  hmrConfig = {
+    protocol: "wss",
```

Comment:
For instance, on my macOS system, 'http://jakemoves-shop.local' is a valid alias for 'localhost', but it will probably die here trying to use `wss`.

Reply:
Probably reasonable to wait and see if we get any issue reports, just wanted to mention this as an edge case.